src/app/middleware.ts

```diff
@@ -25,6 +26,36 @@ class SteamEventMiddleware implements Mountable {
             })
         }
 
+        // hook login state (user login/logout)
+        this.activeHooks.push(SteamClient.User.RegisterForLoginStateChange((username: string) => {
+            if (username) {
+                this.eventBus.emit({
+                    type: "UserLoggedIn",
+                    createdAt: this.clock.getTimeMs(),
+                    username: username
+                });
+            } else {
+                this.eventBus.emit({
+                    type: "UserLoggedOut",
+                    createdAt: this.clock.getTimeMs()
+                })
+            }
+        }));
+
+        // hook app overview changes (not used atm)
+        this.activeHooks.push(SteamClient.Apps.RegisterForAppOverviewChanges((_: ArrayBuffer) => {
+            //logger.debug('AppOverviewChanges proto', _);
+            this.eventBus.emit({
+                type: "AppOverviewChanged",
+                createdAt: this.clock.getTimeMs()
+            });
+        }));
+
+        // patch steam methods so we can overwrite time played
+        let steamPatchesHook = new SteamPatchesHook(this.eventBus, this.clock);
```

Comment:
It is better to create this component in `index.ts` and make `SteamPatchesHook` implement `Mountable`.
```ts
mountManager.addMount(new SteamPatchesHook(this.eventBus, this.clock))
```


MountManager jobs is to handle all thing that needed `mount` and `unmount` events. And this `SteamEventMiddleware` is also using it, so it should not break anything.

Reply:
Gotcha.